### ISSUE TITLE
Typst citations does not work since Pandoc 3.1.9 

### ISSUE THREAD

**Issue by cderv** [quarto-dev/quarto-cli#7586]
Using **knitr** in the example because it is easy to create a bibliography 
````markdown
---
title: "Test"
format: typst
bibliography: packages.bib
---


```{r}
#| include: false
knitr::write_bib("knitr", "packages.bib")
```

See @R-knitr

````

When rendering 
````
[typst]: Compiling index.typ to index.pdf...error: expected label, found string
    ┌─ \\?\C:\Users\chris\Documents\DEV_OTHER\DEMOS\test-quarto\index.typ:239:10
    │
239 │ See #cite("R-knitr")
    │           ^^^^^^^^^
````

This is a known issue with Pandoc 3.1.9 which is not yet compatible with Typst 0.9.0

* https://github.com/jgm/pandoc/issues/9188

It has been solved for next Pandoc version. 

When we do our dependency refresh (https://github.com/quarto-dev/quarto-cli/pull/7419), we should maybe not update typst to the latest version before Pandoc does a new update after typst release... There will probably always be conflict if they keep doing breaking change in typst 

Should we try to path or just wait for Pandoc new release ? 

By the way, it was not catch in test, because it is not done correctly I believe. I'll PR a change that should show the issue, and we could merge new test when solved

**Comment by cscheid** [comment]
We already have a Pandoc 3.1.9 typst workaround for images; we can add an additional one for citations that produces the right syntax. I think I can make this work. Let me give it a shot.

**Comment by cderv** [comment]
>  we can add an additional one for citations that produces the right syntax. I think I can make this work. Let me give it a shot.

yeah I guess we don't when they will do a patch release of Pandoc. 

And yes I think we can make this work by emitting ourself the typescript for `Cite` - we just need to check correctly the `+citation` extension on writer from our Lua.

**Comment by cscheid** [comment]
> I can give it a try too if you want. Just let me know.

Sure, I don't mind if you try it first! Take a look at `render_typst_fixups`.
